### ISSUE TITLE
Clicking on Join and Eco loading Server list is getting worse

### ISSUE THREAD

**Issue by Conchipedia** [StrangeLoopGames/EcoIssues#6069]
**Version:** 0.7.2.4 beta
as you can see the Fps drops to 2 (this is using Steam Overlay) and it still has only just gotten to the part of telling me my Favourites are offline
![snap2020](https://user-images.githubusercontent.com/29383891/37437801-0a4d6702-27e7-11e8-9e02-0da3067a2e35.png)
then Eco Crashed entirely and refused to even give me the message that Windows loves "eco.exe has stopped responding", so quit it entirely by using Task manager
[output_log.txt](https://github.com/StrangeLoopGames/EcoIssues/files/1813585/output_log.txt)

**Comment by MarjieAVolk** [comment]
Yeah, server browser needs some serious TLC

In the log, there's a bunch of ping failures and a couple of these, don't know if that's related:

```
UriFormatException: URI scheme must start with a letter and must consist of one of alphabet, digits, '+', '-' or '.' character.
  at System.Uri.Parse (UriKind kind, System.String uriString) [0x00000] in <filename unknown>:0 
  at System.Uri.ParseUri (UriKind kind) [0x00000] in <filename unknown>:0 
  at System.Uri..ctor (System.String uriString, Boolean dontEscape) [0x00000] in <filename unknown>:0 
  at System.Uri..ctor (System.String uriString) [0x00000] in <filename unknown>:0 
  at System.Net.WebRequest.Create (System.String requestUriString) [0x00000] in <filename unknown>:0 
  at Eco.Shared.Services.ServerBrowser.GetPublicServers (System.Action`1 callback, System.String serversUri) [0x00000] in <filename unknown>:0 
  at Eco.Shared.Services.ServerBrowser.GetPublicServers (System.Action`1 callback, System.String serversUri) [0x00000] in <filename unknown>:0 
  at ServerBrowserUI.Refresh () [0x00000] in <filename unknown>:0 
  at ServerBrowserUI.Start () [0x00000] in <filename unknown>:0 
```

**Comment by GophTheGreat** [comment]
+2 from the support queue, no other valuable info though.

**Comment by Conchipedia** [comment]
i have looked more into this on 7.3.0, if i drag the slider down to the bottom i can still see it is adding to the list and the fps is dropping to single figures, then it is then sorting this list into an order from the lowest ms to the highest.
if the list is left alone and you don't touch the scroll bar the fps goes back up again, but this is also being effected by the Animation that is going on in the background, do we really need this going while we are looking through the server list?

**Comment by MarjieAVolk** [comment]
Yeah in the new version of the server browser, I hope to only have a small number of server listings in the 'public servers' section at a time, maybe 20, with a load more/next page button.  This should hooopefully keep the framerate up.  I am under the impression that the scroll window with tons of items in it is the thing that is slowing everything down.

**Comment by Conchipedia** [comment]
i added to the Above Comment with my output_log.txt as this was added before i saw your reply

**Comment by Conchipedia** [comment]
if you load up a normal version of eco (preferably with steam overlay with fps showing) you watch it drop as soon as you touch that scrollbar

**Comment by Conchipedia** [comment]
i can understand them loading the list, but then it sorts it as well, unless you realise it is doing this it is almost impossible to work out which servers you have been looking at already

**Comment by Conchipedia** [comment]
i tried once to count how many Servers are on the list but i gave up after i counted to over 100 and then suddenly saw the same name appear again lower down 😆
